test/lib/auth_cache/assume_role_web_identity_adapter_test.exs

```diff
@@ -0,0 +1,108 @@
+defmodule ExAws.STS.AuthCache.AssumeRoleWebIdentityAdapterTest do
+  use ExUnit.Case, async: true
+  alias ExAws.STS.AuthCache.AssumeRoleWebIdentityAdapter
+
+  import Mox
+
+  @role_arn "2222222/test_role"
+  @web_identity_token "yo"
+  @role_session_name "default_session"
+  @expiration 30_000
+
+  setup do
+    {:ok, path} = Briefly.create()
+    File.write!(path, @web_identity_token)
+
+    System.put_env("AWS_WEB_IDENTITY_TOKEN_FILE", path)
+    System.put_env("AWS_ROLE_ARN", @role_arn)
+
+    on_exit(fn ->
+      System.delete_env("AWS_WEB_IDENTITY_TOKEN_FILE")
+      System.delete_env("AWS_ROLE_ARN")
+    end)
+  end
+
+  test "injected loader and config" do
```

Comment:
Similar to the comment above, I'd refactor this to use the `test` block to write the expectations (with the MFA prepended), and use a `describe` block to set the context, e.g.:
```elixir
describe "when the loader and config are injected" do
   # ...
end

describe "when the loader and config are the default" do
   # ...
end
```